File: app/src/main/java/com/example/learn/ui/viewswitcher/ViewSwitcherActivity.java
package com.example.learn.ui.viewswitcher;

import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.widget.ViewSwitcher;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.example.learn.R;

import java.util.ArrayList;
import java.util.List;

/**
 * @author fqxyi
 * @date 2020/8/18
 * ViewSwitcher使用示例
 */
public class ViewSwitcherActivity extends AppCompatActivity {

    private int index = 0;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_view_switcher);
        testViewSwitcher();
    }

    private void testViewSwitcher() {
        final ViewSwitcher viewSwitcher = (ViewSwitcher) findViewById(R.id.view_switcher);
        viewSwitcher.setFactory(new ViewSwitcher.ViewFactory() {
            @Override
            public View makeView() {
                return new ViewSwitcherItemView(ViewSwitcherActivity.this);
            }
        });
        viewSwitcher.setInAnimation(this, R.anim.view_switcher_slide_in);
        viewSwitcher.setOutAnimation(this, R.anim.view_switcher_slide_out);

        final List<ViewSwitcherBean> viewSwitcherBeanList = new ArrayList<>();

        for (int i = 0; i < 1000; i++) {
            ViewSwitcherBean viewSwitcherBean = new ViewSwitcherBean();
            viewSwitcherBean.title = "title " + i;
            viewSwitcherBean.content = "content " + i;
            viewSwitcherBeanList.add(viewSwitcherBean);
        }

        ViewSwitcherItemView viewSwitcherItemView = (ViewSwitcherItemView) viewSwitcher.getNextView();
        viewSwitcherItemView.initData(viewSwitcherBeanList.get(index++ % viewSwitcherBeanList.size()));
        viewSwitcher.showNext();

        final Handler handler = new Handler();
        handler.postDelayed(new Runnable() {
            @Override
            public void run() {
                ViewSwitcherItemView viewSwitcherItemView = (ViewSwitcherItemView) viewSwitcher.getNextView();
                viewSwitcherItemView.initData(viewSwitcherBeanList.get(index++ % viewSwitcherBeanList.size()));

                if (index >= viewSwitcherBeanList.size()) {
                    index = 0;
                }

                viewSwitcher.showNext();

                handler.postDelayed(this, 2000);
            }
        }, 2000);
    }
}

File: app/src/main/java/com/example/learn/ui/viewswitcher/ViewSwitcherActivity.kt
package com.example.learn.ui.viewswitcher

import android.os.Bundle
import android.os.Handler
import com.example.common.base.BaseActivity
import com.example.learn.R
import kotlinx.android.synthetic.main.activity_view_switcher.*
import java.util.*

/**
 * @author fqxyi
 * @date 2020/8/18
 * ViewSwitcher使用示例
 */
class ViewSwitcherActivity : BaseActivity() {

    private var index = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_view_switcher)
        testViewSwitcher()
    }

    private fun testViewSwitcher() {
        viewSwitcher.setFactory { ViewSwitcherItemView(this@ViewSwitcherActivity) }
        viewSwitcher.setInAnimation(this, R.anim.view_switcher_slide_in)
        viewSwitcher.setOutAnimation(this, R.anim.view_switcher_slide_out)
        val list = ArrayList<ViewSwitcherBean>()
        for (i in 0..999) {
            list.add(ViewSwitcherBean("title $i", "content $i"))
        }
        val itemView = viewSwitcher.nextView as ViewSwitcherItemView
        itemView.initData(list[index++ % list.size])
        viewSwitcher.showNext()
        val handler = Handler()
        handler.postDelayed(object : Runnable {
            override fun run() {
                val innerItemView = viewSwitcher.nextView as ViewSwitcherItemView
                innerItemView.initData(list[index++ % list.size])
                if (index >= list.size) {
                    index = 0
                }
                viewSwitcher.showNext()
                handler.postDelayed(this, 2000)
            }
        }, 2000)
    }
}